apps/dot_ext/forms.py

```diff
@@ -120,12 +130,28 @@ def clean_redirect_uris(self):
                         raise forms.ValidationError(msg)
         return redirect_uris
 
+    def clean_logo_image(self):
+        logo_image = self.cleaned_data.get('logo_image')
+        if getattr(logo_image, 'name', False):
+            validate_logo_image(logo_image)
+        return logo_image
+
     def save(self, *args, **kwargs):
         app = self.instance
         logmsg = "%s agreed to %s for the application %s" % (app.user, app.op_tos_uri,
                                                              app.name)
         logger.info(logmsg)
-        return super().save(*args, **kwargs)
+        app = super().save(*args, **kwargs)
+        logo_image = self.cleaned_data.pop('logo_image', None)
+        if getattr(logo_image, 'name', False):
+            file_path = "applications/" + hashlib.sha256(str(app.pk).encode('utf-8')).hexdigest() + "/logo.jpg"
+            if default_storage.exists(file_path):
+                default_storage.delete(file_path)
+            default_storage.save(file_path, logo_image)
+            if default_storage.exists(file_path):
+                app.logo_uri = settings.MEDIA_URL + file_path
+            app.save()
```

Comment:
I'd like this upload logic to be split out into something that's only responsible for handling the logo_image/logo_uri. Could be as simple as a separate function that's called from here.

Reply:
I like this. Will update shortly.